packages/uniforms-bootstrap4/src/SelectField.tsx

```diff
@@ -87,22 +87,38 @@ function Select({
         })}
         disabled={disabled}
         id={id}
+        multiple={fieldType === Array}
         name={name}
-        onChange={event =>
-          onChange(event.target.value !== '' ? event.target.value : undefined)
-        }
+        onChange={event => {
+          if (fieldType === Array) {
+            return;
+          }
+          const item = event.target.value;
+          onChange(item !== '' ? item : undefined);
+        }}
         ref={inputRef}
         value={value ?? ''}
       >
-        {(!!placeholder || !required || value === undefined) && (
-          <option value="" disabled={required} hidden={required}>
-            {placeholder || label}
-          </option>
-        )}
+        {(!!placeholder || !required || value === undefined) &&
+          fieldType !== Array && (
+            <option value="" disabled={required} hidden={required}>
+              {placeholder || label}
+            </option>
+          )}
 
-        {allowedValues?.map(value => (
-          <option disabled={disableItem?.(value)} key={value} value={value}>
-            {transform ? transform(value) : value}
+        {allowedValues?.map(allowedValue => (
+          <option
+            disabled={disableItem?.(allowedValue)}
+            key={allowedValue}
+            value={allowedValue}
+            onClick={event => {
+              if (fieldType === Array) {
+                const item = event.currentTarget.value;
+                onChange(fieldType === Array ? xor([item], value) : item);
+              }
```

Comment:
I think that `onClick` here is not the best idea - it may have problems on phones (probably) as well as when used with keyboard (definitely). As I remember, the problem was to detect if nothing is selected. Try out the [`selectedIndex`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/selectedIndex) property, that is `-1` if nothing is selected.